Client/Frontend/Browser/FrequencyQuery.swift

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import Storage
+import Data
+
+class FrequencyQuery {
+    
+    private static let queue = DispatchQueue(label: "frequency-query-queue")
+    private static var cancellable: DispatchWorkItem?
+    
+    public static func sitesByFrequency(containing query: String? = nil,
+                                        completion: @escaping (Set<Site>) -> Void) {
+        
+        Historyv2.byFrequency(query: query) { historyList in
+            let historySites = historyList
+                .map { Site(url: $0.url ?? "", title: $0.title ?? "") }
+            
+            cancellable = DispatchWorkItem {
+                // brave-core fetch can be slow over 200ms per call,
+                // a cancellable serial queue is used for it.
+                DispatchQueue.main.async {
+                    let bookmarkSites = Bookmarkv2.byFrequency(query: query)
```

Comment:
Does the Brave-Core that History Sync is in, also contain the Bookmark `byFrequency` fix which uses completion block like `History.byFrequency` OR will that be next release?

Reply:
I forgot to uplift your fix to 1.26.x. Thats why we dont have the change here but when we fetch core 1.27+, we need to make a change here with it.